### PR TITLE
[WB-4896] Fixing Audio Reference and Making Artifact Test Better

### PR DESCRIPTION
**Pull request by tssweeney**
https://wandb.atlassian.net/browse/WB-4896

Description
-----------

Allows artifacts to resolve local paths to named entries and uses that functionality to fix audio reference bug detected by the regression suite: https://wandb.atlassian.net/browse/WB-4896. Also, added more debuggablity to regression suite.

Testing
-------

How was this PR tested?

### CODE REVIEW

**Comment by tssweeney** [comment]
Local `artifact_object_reference_test.py ` completes end to end

File: standalone_tests/artifact_object_reference_test.py [hunk 1]
```diff
@@ -389,6 +386,7 @@ def test_get_artifact_obj_by_name():
         assert actual_table.columns == columns
         assert actual_table.data[0][columns.index("Image")] == image
         assert actual_table.data[1][columns.index("Image")] == _make_wandb_image("2")
+        actual_table._eq_debug(_make_wandb_table(), True)
```
Comment (davidwallacejackson) [inline standalone_tests/artifact_object_reference_test.py:389]: Are these debug calls still supposed to be here? I thought they were for dropping in temporarily on a misbehaving test.

Comment (tssweeney) [reply standalone_tests/artifact_object_reference_test.py:389]: yes, since they will give a better error description.

Comment (davidwallacejackson) [reply standalone_tests/artifact_object_reference_test.py:389]: Got it. What about the `==` below, then -- is that redundant?

**Review by davidwallacejackson — APPROVED** [review]

File: standalone_tests/artifact_object_reference_test.py (hunk 1 above)
Comment (tssweeney) [reply standalone_tests/artifact_object_reference_test.py:389]: well i figured there is a non zero chance something get borked in implementation between _eq_debug and __eq__ so i kept both for good measure.

**Pull request merged**